js/src/ui/TxList/TxRow/txRow.js

```diff
@@ -137,11 +142,46 @@ class TxRow extends Component {
     return (
       <td className={ styles.timestamp }>
         <div>{ blockNumber && block ? moment(block.timestamp).fromNow() : null }</div>
-        <div>{ blockNumber ? _blockNumber.toFormat() : 'Pending' }</div>
+        <div>{ blockNumber ? _blockNumber.toFormat() : this.renderCancelToggle() }</div>
       </td>
     );
   }
 
+  renderCancelToggle () {
+    const { isCancelOpen, canceled } = this.state;
+    const { parity } = this.context.api;
+    const { hash } = this.props.tx;
+
+    if (canceled) {
+      return (
+        <div className={ styles.pending }>CANCELED</div>
+      );
+    }
+
+    if (!isCancelOpen) {
+      return (
+        <div className={ styles.pending }>
+          <div>SCHEDULED</div>
+          <a
+            className={ styles.cancel }
+            onClick={ () => this.setState({ isCancelOpen: true }) }
```

Comment:
Would prefer not using in-line closures, aligning with the codebase. (I actually thought we had a linting rule that warns about these, weird that it wasn't picked up)